candies/cli/cli.py

```diff
@@ -57,6 +61,60 @@ def help(self, *args, **kwargs):
         """
         return self.main.help(*args, **kwargs)
 
+    def parse(self, callable: Union[Callable[[str], Any], Type]) \
+            -> Union[Type, Callable[[str], Any]]:
+        """Defines a parser for a custom type.
+
+        Args:
+            callable: A callable that converts a string to an instance
+                of a custom type. For example, a parser function or a
+                type that accepts a string to a constructor.
+
+        Examples:
+            @cli
+            def add(money: Arg[Money]):
+                ...
+
+            @main.parse
+            def money(x: str) -> Money:
+                if x[0] == '$':
+                    return Money(float(x[1:]), 'dollars')
+                ...
+
+            # Or, as an alternative, if
+            # `Money.__init__(str)` can be called.
+            main.parse(Money)
```

Comment:
Well, seems like `@main.parse` looks good, when decorating a function, but here I'd rather expect the `main.register_parser_for(Money)` call. Though have no idea what to do. Should we point this way of doing things as not recommended? Or is there a better alternative for `parse` name?

Reply:
That's a good observation. I think we could stick with `register_parser_for` for now.

Reply:
Maybe `add_parser` would be a shorter version for `register_parser_for`. Also I think that maybe it'd be better to support both `parse` (or `parser`) and `add_parser`, because `@main.add_parser` looks not that good as `@main.parser` or `@main.parse`.

Reply:
Added the `add_parser` method. Thanks!